packages/server-reason-react-ppx/server_reason_react_ppx.ml

```diff
@@ -765,6 +802,34 @@ let raise_usage_of_client_components module_expr =
         ]
   | _ -> module_expr
 
+(*
```

Comment:
@davesnx I created this transform to ensure props are declared with [react.server.action] that it will be `string` in the end as we pass IDs on Native instead of the function.
```
Native: React.Action(string)
Js: (...'a) => Js.Promise.t('b)
```

```reason
[@react.client.component]
let make =
    (~foo: [@react.server.action] (unit => Js.Promise.t(string))) => {

  //...
};

// Native
[@react.client.component]
let make =
    (~foo: [@react.server.action] string) => {

  //...
};

// Js
[@react.client.component]
let make =
    (~foo: [@react.server.action] (unit => Js.Promise.t(string))) => {

  //...
};
```
Do you have another idea for it?